### PR TITLE
[CLN] parametrize and cleanup a bunch of tests

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -74,29 +74,29 @@ def test_corr_non_numeric(self):
         tm.assert_frame_equal(result, expected)
 
     @td.skip_if_no_scipy
-    def test_corr_nooverlap(self):
+    @pytest.mark.parametrize('meth', ['pearson', 'kendall', 'spearman'])
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: should make this a fixture

File: pandas/tests/frame/test_analytics.py [hunk 2]
```diff
@@ -658,21 +658,21 @@ def test_numeric_only_flag(self, meth):
         pytest.raises(TypeError, lambda: getattr(df2, meth)(
             axis=1, numeric_only=False))
 
-    def test_mixed_ops(self):
+    @pytest.mark.parametrize('op', ['mean', 'std', 'var',
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: do we have these as fixtures?

File: pandas/tests/frame/test_axis_select_reindex.py [hunk 3]
```diff
@@ -674,29 +674,29 @@ def _check_align(self, a, b, axis, fill_axis, how, method, limit=None):
...
-                for fax in [0, 1]:
-                    self._check_align_fill('left', meth, ax, fax)
-
-    def test_align_fill_method_right(self):
-        for meth in ['pad', 'bfill']:
-            for ax in [0, 1, None]:
-                for fax in [0, 1]:
-                    self._check_align_fill('right', meth, ax, fax)
+    @pytest.mark.parametrize('meth', ['pad', 'bfill'])
+    @pytest.mark.parametrize('ax', [0, 1, None])
```
Comment (jreback) [inline pandas/tests/frame/test_axis_select_reindex.py]: we have the axis arg as a fixture (though may need to add another one which adds None to it)

File: pandas/tests/frame/test_axis_select_reindex.py [hunk 4]
```diff
@@ -674,29 +674,29 @@ def _check_align(self, a, b, axis, fill_axis, how, method, limit=None):
...
+    def test_align_fill_method_outer(self, meth, ax, fax):
+        self._check_align_fill('outer', meth, ax, fax)
+
+    @pytest.mark.parametrize('meth', ['pad', 'bfill'])
+    @pytest.mark.parametrize('ax', [0, 1, None])
+    @pytest.mark.parametrize('fax', [0, 1])
+    def test_align_fill_method_left(self, meth, ax, fax):
+        self._check_align_fill('left', meth, ax, fax)
+
+    @pytest.mark.parametrize('meth', ['pad', 'bfill'])
```
Comment (jreback) [inline pandas/tests/frame/test_axis_select_reindex.py]: maybe add as a fixture (if these are named appropriately they immeditaly become obvious as fixtures)

File: pandas/tests/frame/test_reshape.py [hunk 5]
```diff
@@ -855,21 +855,21 @@ def _test_stack_with_multiindex(multiindex):
                              dtype=df.dtypes[0])
         assert_frame_equal(result, expected)
 
-    def test_stack_preserve_categorical_dtype(self):
+    @pytest.mark.parametrize('ordered', [False, True])
```
Comment (jreback) [inline pandas/tests/frame/test_reshape.py]: I think ordered is a fixture already

File: pandas/tests/scalar/period/test_period.py [hunk 6]
```diff
@@ -1038,9 +1038,10 @@ def test_add_raises(self):
             dt1 + dt2
 
     boxes = [lambda x: x, lambda x: pd.Series([x]), lambda x: pd.Index([x])]
```
Comment (jreback) [inline pandas/tests/scalar/period/test_period.py]: maybe move these 2 lines to the top of the file?

File: pandas/tests/series/test_alter_axes.py [hunk 7]
```diff
@@ -245,7 +245,7 @@ def test_set_axis_inplace(self):
         expected = s.copy()
         expected.index = list('abcd')
 
-        for axis in 0, 'index':
+        for axis in [0, 'index']:
```
Comment (jreback) [inline pandas/tests/series/test_alter_axes.py]: axis_series

File: pandas/tests/series/test_operators.py [hunk 8]
```diff
@@ -1587,7 +1606,25 @@ def test_operators_reverse_object(self, op):
         expected = op(1., arr.astype(float))
         assert_series_equal(result.astype(float), expected)
 
-    def test_operators_combine(self):
+    pairings = []
+    for op in ['add', 'sub', 'mul', 'pow', 'truediv', 'floordiv']:
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: I think we already have these no as fixtures?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
comments and pls rebase

File: pandas/tests/series/test_operators.py (hunk 8 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: conftest.py doesn't have one specifically for this subset.  Figuring out what is worth fixturizing will be much easier after arithmetic tests are centralized (#22033 and follow-ups)

File: pandas/tests/scalar/period/test_period.py (hunk 6 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/period/test_period.py]: If we use them more than once it makes sense, but ATM they are defined right next to where they are used, which is pretty ideal.

File: pandas/tests/frame/test_reshape.py (hunk 5 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_reshape.py]: Not in conftest.

File: pandas/tests/frame/test_analytics.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_analytics.py]: Not AFAICT

File: pandas/tests/series/test_operators.py (hunk 8 above)
Comment (jreback) [reply pandas/tests/series/test_operators.py]: sure we do, all_arithmetic_operators (and several slices of this)

File: pandas/tests/frame/test_reshape.py (hunk 5 above)
Comment (jreback) [reply pandas/tests/frame/test_reshape.py]: pandas/tests//arrays/categorical/conftest.py

maybe can simply move it to top level and use it

File: pandas/tests/frame/test_analytics.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/frame/test_analytics.py]: cython_table_items yields these (well it yields them all, maybe need to make a subset)

File: pandas/tests/frame/test_reshape.py (hunk 5 above)
Comment (gfyoung) [reply pandas/tests/frame/test_reshape.py]: Honestly, I would love to create a helper in `utils/testing.py` to generate boolean fixtures like these on the fly.  Not sure if it's possible, but just throwing it out there for thought.

File: pandas/tests/frame/test_analytics.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_analytics.py]: > well it yields them all, maybe need to make a subset

Consider:

```
@pytest.mark.parametrize('op', SPECIFIC_EXPLICIT_LIST)
def test_thing(self, op):

def test_thing(self, cython_table_items):
     if cython_table_items not in SPECIFIC_EXPLICIT_LIST:
          return
```

There are good use cases for fixtures, this isn't near the top of that list.

File: pandas/tests/series/test_operators.py (hunk 8 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: See above re subsetting fixtures.

File: pandas/tests/frame/test_axis_select_reindex.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/frame/test_axis_select_reindex.py]: ok for sure need to use the fixture (but again can come back later on this)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**